content/docs/rfcs/35/README.md

```diff
@@ -0,0 +1,320 @@
+
+In particular, if `protocol-id` is
+- `0`:  payload encodes an [after-handshake](#After-handshake) message.  
+    - `handshake-message-len` MAY be 0;
+    -  `transport-message` contains the Noise transport message;
+- `10`,`11`,`12`,`13`: payload encodes a supported Noise handshake message.  
+    -  `transport-message` contains the Noise transport message;
+- `30`: payload encapsulate a `ChaChaPoly` ciphertext `ct`.  
+    - `handshake-message-len` is set to `0`;
+    - `transport-message` contains the concatenation of the encryption nonce (12 bytes) followed by the ciphertext `ct`;
+    - `transport-message-len-len` and `transport-message-len` are set accordingly to `transport-message` length;
+    - `transport-message-ad` contains the authentication data for `ct`.
+
+
+### Public Keys Serialization
+
+Diffie-Hellman public keys can be trasmitted in clear 
+or in encrypted form with authentication data attached. 
+To distinguish between these two cases, public keys are serialized as the concatenation of the following three fields:
+
+- `flag`: 
+is equal to `1` if the public key is encrypted; 
+`0` otherwise (1 byte);
+- `pk`: 
+if `flag = 0`, it contains an encoding of the X coordinate of the public key. 
+If `flag = 1`, it contains a symmetric encryption of an encoding of the X coordinate of the public key;
+- `pk_ad`: 
+if `flag = 0`, it is empty; 
+if `flag = 1`, it contains the authentication data for `pk`;
+
+
+As regards the underlying supported [cryptographic primitives](#Cryptographic-primitives):
+- `Curve25519` public keys X coordinates are encoded in little-endian as 32 bytes arrays;
+- `ChaChaPoly` authentication data consists of 16 bytes 
+(nonces are implicitely defined by Noise [processing rules](http://www.noiseprotocol.org/noise.html#processing-rules)).
+
+In all supported Noise protocols, 
+parties' static keys are transmitted encrypted, 
+while ephemeral keys MAY be encrypted after a handshake is complete.
+
+
+### Padding
+
+To prevent some metadata leakage, 
+encrypted transport messages SHOULD be padded before encryption 
+to a multiple of the underlying symmetric cipher block size 
+(16 bytes for `ChaChaPoly`).
+
+It is therefore recommended to pad transport messages to a multiple of 256 bytes.
+
+
+## After-handshake
+
+We note that 
+during the initial 1-RTT communication, 
+handshake messages can be linked to the respective parties 
+through the `contentTopic` employed for such communication. 
+
+After a handshake is completed, 
+parties SHOULD derive from their shared secret key (preferably using `HKDF`) a new random `contentTopic` 
+and continue their communication there.
+
+When communicating on the new `contentTopic`, 
+parties SHOULD set `protocol-id` to `0` 
+to reduce metadata leakages and indicate that the message is an *after-handshake* message.
+
+It is recommended that each party attaches an (unencrypted) ephemeral key in `handshake-message` to every message sent.
```

Comment:
We usually upper case the RFC-2119 terminology: https://rfc.vac.dev/spec/1/#language

nitpick: Active voice is usually easier to read than passive voice (if my grammar is correct)

```suggestion
Each party SHOULD attach an (unencrypted) ephemeral key in `handshake-message` to every message sent.
```

Reply:
Addressed in https://github.com/vacp2p/rfc/pull/496/commits/b587e8b2cb05ce1fa6042c2417696cb50be6fcf4